ios/browser/api/brave_shields/adblock_service.mm

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/brave_shields/adblock_service.h"
+
+#include "base/strings/sys_string_conversions.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/adblock_rust_ffi/src/wrapper.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/brave_shields/browser/ad_block_component_installer.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/ios/browser/api/brave_shields/adblock_filter_list+private.h"
+#include "components/component_updater/component_updater_service.h"
+
+@interface AdblockService () {
+  component_updater::ComponentUpdateService* _cus;  // NOT OWNED
+}
+@property(nonatomic, copy) NSString* shieldsInstallPath;
+@property(nonatomic, copy) NSArray<AdblockFilterList*>* regionalFilterLists;
+@end
+
+@implementation AdblockService
+
+- (instancetype)initWithComponentUpdater:
+    (component_updater::ComponentUpdateService*)componentUpdaterService {
+  if ((self = [super init])) {
+    _cus = componentUpdaterService;
+  }
+  return self;
+}
+
+- (void)registerDefaultShieldsComponent {
+  __weak auto weakSelf = self;
+  brave_shields::RegisterAdBlockDefaultComponent(
+      _cus, base::BindRepeating(^(const base::FilePath& install_path) {
+        [weakSelf adblockComponentDidBecomeReady:install_path];
+      }));
+}
+
+- (void)adblockComponentDidBecomeReady:(const base::FilePath&)install_path {
+  auto component_path = install_path;
```

Comment:
`nit:` If you change `const base::FilePath&` to `base::FilePath`, you'll be able to `install_path.AppendASCII("regional_catalog.json")` without having to make a copy here as the compiler will take care of whether it needs a copy or not :D

Reply:
Done in https://github.com/brave/brave-core/pull/14248/commits/3d88a54b96aa61395da05e7196b98d3fc3337165